db/fetcher/fetcher.go

```diff
@@ -31,6 +31,8 @@ import (
 // It handles all the key/value scanning, aggregation, and document encoding.
 type Fetcher interface {
 	Init(
+		ctx context.Context,
+		txn datastore.Txn,
```

Comment:
question: so we now have `txn` for `Init` and `Start`. How are they related? Does it have to be the same transaction? Does it ever make sense to to provide different transactions for these methods? If not maybe we should remove `txn` from `Start`.